compiler/rustc_monomorphize/src/partitioning.rs

```diff
@@ -680,6 +687,16 @@ fn compute_codegen_unit_name<'tcx>(
     mono_item: MonoItem<'tcx>,
     cache: &mut CguNameCache,
 ) -> Symbol {
+    // When compiling compiler_builtins, we do not want to put multiple intrinsics in a CGU.
+    // Using the symbol name as the CGU name puts every GloballyShared item in its own CGU, but in
+    // an optimized build we actually want every item in the crate that isn't an intrinsic to get
+    // LocalCopy so that it is easy to inline away. In an unoptimized build, this CGU naming
+    // strategy probably generates more CGUs than we strictly need. But it is simple.
+    if tcx.is_compiler_builtins(LOCAL_CRATE) {
+        let name = mono_item.symbol_name(tcx);
+        return Symbol::intern(name.name);
```

Comment:
One of the symbols in compiler-builtins is 132 characters long, together with the crate name and the temporary directory, this could exceed MAX_PATH on Windows I think. Maybe hash the name if its length exceeds say 50 characters